mjtest-files/compile/bench/SubsetSum.mj

```diff
@@ -0,0 +1,193 @@
+class A {
+    public int[] values;
+    public int num_rows;
+    public boolean[][] permutations;
+    public int[] solutionsPerValue;
+
+   public static void main(String[] args) {
+        A obj = new A();
+        obj.run();
+    }
+
+    public void run() {
+        int n = 30;
+        init(n);
+
+        /* we do repeated runs to get an appropriate running time */
+        int i = 0;
+        int sum = 0;
+        while (i < 5) {
+            sum = sum + runSingle(i, n);
+            i = i + 1;
+        }
+        System.out.println(sum);
+    }
+
+    public int runSingle(int target, int n) {
+        int row = 0;
+        while (row < num_rows) {
+            int i = 0;
+            while (!permutations[row][i]) {
+                i = i + 1;
+            }
+            int a = values[i];
+            i = i + 1;
```

Comment:
couldnt you declare new variables here to increase register pressure?

Reply:
That's not the purpose of the benchmark, and declaring new variables doesn't help anyway if they are just optimized away in the middle end.